Created Matplotlib directory and added a term for legend (#923)
Codecademy/docs: content/python/concepts/matplotlib/terms/legend/legend.md

---
Title: '.legend()'
Description: 'Places a legend on the axes and access to manipulate parameters of the legend.'
Subjects:
  - 'Data Science'
  - 'Data Visualization'
Tags:
  - 'Graphs'
  - 'Libraries'
  - 'Matplotlib'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

The **`pyplot.legend()`** function returns a legend on the axes of a plot or subplot.

## Syntax

```pseudo
plot_instance.legend(handles, labels)
```

> The `.legend()` accepts zero or more parameters and can be called on a valid `plot_instance` of the `pyplot` module. When called with no parameters, the `.legend()` function will attempt to add the elements automatically. The following parameters can be used in the `.legend()` function:

Reviewer: It's okay to refer to `pyplot.legend()` as a function, but the way it's mentioned here as `.legend`, it should be referred to as a method

|  Parameter Name  | Data Type(s)         | Usage                                                                                                  |
| :--------------: | -------------------- | ------------------------------------------------------------------------------------------------------ |
|    `handles`     | sequence of `Artist` | `handles` is utilized if more control is desired with each entry on the plot and how they are labeled. |

> |     `labels`     | list of strings       | `labels` is a list of strings that will display the labels in the order they are entered.              |
> |      `loc`       | int, string          | `loc` is used to determine the position of the legend on the axes. Defaults to `best`.                 |
> |      `ncol`      | int                  | `ncol` is used to determine how many columns the legend will have. Defaults to 1.                      |
> |    `fontsize`    | int, string          | `fontsize` is used change the font size the labels. Defaults to 11.                                    |
> |   `labelcolor`   | list, string, `None` | `labelcolor` is used to change the color of the label(s). Defaults to `None`                           |
> |     `title`      | string, `None`       | `title` is used to insert a title for the legend. Defaults to `None`.                                  |
> | `title_fontsize` | int, string, `None`  | `title_fontsize` is used to change the font size of the legend `title`. Defaults to `None`.            |

Reviewer: ```suggestion
|     `labels`     | list of strings       | A list of strings that will display the labels in the order they are entered.              |
|      `loc`       | int, string          | Used to determine the position of the legend on the axes. Defaults to `best`.                 |
|      `ncol`      | int                  | Used to determine how many columns the legend will have. Defaults to 1.                      |
|    `fontsize`    | int, string          | Used change the font size the labels. Defaults to 11.                                    |
|   `labelcolor`   | list, string, `None` | Used to change the color of the label(s). Defaults to `None`                           |
|     `title`      | string, `None`       | Used to insert a title for the legend. Defaults to `None`.                                  |
| `title_fontsize` | int, string, `None`  | Used to change the font size of the legend `title`. Defaults to `None`.            |
```

## Example

> In the following example, the `.legend()` method will be shown using the iris data set which is available using seaborn or pandas. In the first example the `handles` parameter will be demonstrated along with using seaborn to import the dataset. In the second example `legend()` will be demonstrated without using the `handles` parameter along with using pandas to import the dataset:

Reviewer: ```suggestion
In the following example, the `.legend()` method will be shown using the iris data set which is available using `seaborn` or [`pandas`](https://www.codecademy.com/resources/docs/pandas). In the first example, the `handles` parameter will be demonstrated, using `seaborn` to import the dataset. In the second example, `legend()` will be demonstrated without using the `handles` parameter along, using pandas to import the dataset:
```

Seaborn can be added as a link after the `seaboarn` entry is added!

Reviewer: @cjunhyuk is there a reason we want two examples? If so, can we include the reasoning here as to why we would use `seaborn` over `pandas` or vice versa?

Author: The first example is a base case with no alterations to any of the parameters for the legend. The second example is demonstrating some of the parameters and showing it. 
The use of `pandas` compared to `seaborn` is negligible in this instance. Since I was showing two examples I just showed two different ways to accessing the same built in dataset.
We can use only 1 example if that would be preferred.

---
Title: '.legend()'
Description: 'Places a legend on the axes and access to manipulate parameters of the legend.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Data Visualization'
Tags:
  - 'Libraries'
  - 'Matplotlib'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> The **`.legend()`** function returns a legend on the axes of a plot or subplot.

Reviewer 2: ```suggestion
The **`pyplot.legend()`** function returns a legend on the axes of a plot or subplot.
```

I think it's important to include the module that this function came from so it won't be [confused](https://matplotlib.org/stable/search.html?q=legend) with something else in the Matplotlib ecosystem.

## Syntax

```pseudo

> plt.legend(handles, labels)

Reviewer 2: ```suggestion
plot_instance.legend(handles, labels)
```

```

> `legend()` can be called without any parameters. This will simply attempt to add the elements automatically. Both `handles` and `labels` are optional. `handles` is utilized if more control is desired with each entry on the plot and how they are labeled. `labels` is a list of string(s) that will display the labels in the order that they are entered with no regard to the actual entries on the plot.
> 
> `.legend()` has the following parameters:

Reviewer 2: ```suggestion
The `.legend()` accepts zero or more parameters and can be called on a valid `plot_instance` of the `pyplot` module. When called with no parameters, the `.legend()` function will attempt to add the elements automatically. The following parameters can be used in the `.legend()` function:
```

A few additional thoughts:

- The sentence "Both `handles` and `labels` are optional." seems confusing to me. If `.legend()` can be called with no parameters, doesn't that mean that _all_ of the parameters mentioned in the table are optional?

- I would remove the following text and work it into the appropriate table rows below: "`handles` is utilized if more control is desired with each entry on the plot and how they are labeled. `labels` is a list of string(s) that will display the labels in the order that they are entered with no regard to the actual entries on the plot."

|  Parameter Name  | Data Type(s)         | Usage                                                                                                  |
| :--------------: | -------------------- | ------------------------------------------------------------------------------------------------------ |
|    `handles`     | sequence of `Artist` | `handles` is utilized if more control is desired with each entry on the plot and how they are labeled. |

> |     `labels`     | list of string       | `labels` is a list of strings that will display the labels in the order they are entered.              |

Reviewer 2: ```suggestion
|     `labels`     | list of strings       | `labels` is a list of strings that will display the labels in the order they are entered.              |
```

|      `ncol`      | int                  | `ncol` is used to determine how many columns the legend will have. Defaults to 1.                      |
|    `fontsize`    | int, string          | `fontsize` is used change the font size the labels. Defaults to 11.                                    |
|   `labelcolor`   | list, string, `None` | `labelcolor` is used to change the color of the label(s). Defaults to `None`                           |
|     `title`      | string, `None`       | `title` is used to insert a title for the legend. Defaults to `None`.                                  |
| `title_fontsize` | int, string, `None`  | `title_fontsize` is used to change the font size of the legend `title`. Defaults to `None`.            |

## Example

In the following example, the `.legend()` method will be shown using the iris data set which is available using seaborn or pandas. In the first example the `handles` parameter will be demonstrated along with using seaborn to import the dataset. In the second example `legend()` will be demonstrated without using the `handles` parameter along with using pandas to import the dataset:

### Example 1

---

> 

Reviewer: Same here, can we refer to it as a method. Also, put `pandas` in backpacks

```py
import matplotlib.pyplot as plt
import seaborn as sns

iris = sns.load_dataset('iris')

fig, ax = plt.subplots(figsize=(12,6))

sepal_lw = ax.scatter(iris['sepal_length'], iris['sepal_width'], label='Sepal')
petal_lw = ax.scatter(iris['petal_length'], iris['petal_width'], label='Petal')

plt.legend(handles=[sepal_lw, petal_lw]);
```

The output from this instance of the `.legend()` method is shown below:

> ![Example1](/media/matplotlib-legend-1.png)

Reviewer: I'd go with the example that is considered a better practice. If either modules are a good practice to use, and you wish to keep both examples, I'd just go into what the difference is. Maybe one is better for a scenario.

> ![Example1](/media/matplotlib-legend-1.png)

Reviewer: Alternatively, instead of including pandas here, you can cite the existing entries:

- https://www.codecademy.com/resources/docs/pandas/dataframe
OR
- https://www.codecademy.com/resources/docs/pandas/built-in-functions/read-csv
